dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,583 @@
+                             // this is a host
+            host = APILocator.getHostAPI().find(browserQuery.hostFolderId, browserQuery.user, false);
+
+            if (host == null) {
+                Logger.error(this, "Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId
+                                + ", maybe the Folder was modified in the background.");
+                throw new NotFoundInDbException("Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId);
+            }
+        }
+
+        if (browserQuery.showFolders) {
+            if (parent != null) {
+                List<Folder> folders = new ArrayList<Folder>();
+                try {
+                    folders = folderAPI.findSubFolders(parent, userAPI.getSystemUser(), false);
+                } catch (Exception e1) {
+                    Logger.error(this, "Could not load folders : ", e1);
+                }
+                for (Folder folder : folders) {
+                    List<Integer> permissions = permissionAPI.getPermissionIdsFromRoles(folder, roles, browserQuery.user);
+                    if (permissions.contains(PERMISSION_READ)) {
+                        Map<String, Object> folderMap = folder.getMap();
+                        folderMap.put("permissions", permissions);
+                        folderMap.put("parent", folder.getInode());
+                        folderMap.put("mimeType", "");
+                        folderMap.put("name", folder.getName());
+                        folderMap.put("title", folder.getName());
+                        folderMap.put("description", folder.getTitle());
+                        folderMap.put("extension", "folder");
+                        folderMap.put("hasTitleImage", "");
+                        folderMap.put("__icon__", "folderIcon");
+                        returnList.add(folderMap);
+                    }
+                }
+            }
+        }
+
+        if (browserQuery.showLinks) {
+
+            // Getting the links directly under the parent folder or host
+            List<Link> links = new ArrayList<Link>();
+
+            if (parent != null) {
+                if (browserQuery.showWorking) {
+                    links.addAll(folderAPI.getLinks(parent, true, false, browserQuery.user, false));
+                } else {
+                    links.addAll(folderAPI.getLiveLinks(parent, browserQuery.user, false));
+                }
+                if (browserQuery.showArchived)
+                    links.addAll(folderAPI.getLinks(parent, true, browserQuery.showArchived, browserQuery.user, false));
+            } else {
+                links = folderAPI.getLinks(host, true, browserQuery.showArchived, browserQuery.user, false);
+            }
+
+
+            for (Link link : links) {
+
+                List<Integer> permissions2 = permissionAPI.getPermissionIdsFromRoles(link, roles, browserQuery.user);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'permissions2' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)